contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -708,10 +708,24 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
   /// @notice Returns the DON specific configuration for a capability
   /// @param donId The DON's ID
   /// @param capabilityId The Capability ID
-  /// @return bytes The DON specific configuration for the capability
-  function getDONCapabilityConfig(uint32 donId, bytes32 capabilityId) external view returns (bytes memory) {
+  /// @return bytes The DON specific configuration for the capability stored on the capability registry
+  /// @return bytes The DON specific configuration stored on the capability's configuration contract
+  function getDONCapabilityConfig(
+    uint32 donId,
+    bytes32 capabilityId
+  ) external view returns (bytes memory, bytes memory) {
     uint32 configCount = s_dons[donId].configCount;
-    return s_dons[donId].config[configCount].capabilityConfigs[capabilityId];
+    DONCapabilityConfig storage donCapabilityConfig = s_dons[donId].config[configCount];
+
+    bytes memory capabilityConfig = donCapabilityConfig.capabilityConfigs[capabilityId];
+    bytes memory capabilityConfigContractConfig;
+
+    if (bytes32(capabilityConfig) != bytes32("") && s_capabilities[capabilityId].configurationContract != address(0)) {
```

Comment:
```suggestion
    if (s_capabilities[capabilityId].configurationContract != address(0)) {
```

The first check should not be done? There might be a Capability config that's shared across DONs but not a DON-specific config.